### ISSUE TITLE
PrintF not working correctly on mingw64

### ISSUE THREAD

**Issue by cwfitzgerald** [tomtom-international/cpp-dependencies#10]
While compiling went smoothly, g++ threw lots of warnings saying that it had "unknown conversion type character"s in a lot of the printf statements. Here are the errors:

```
$ make
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/main.o src/main.cpp -std=c++11 -O3
src/main.cpp: In lambda function:
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
                     components.size(), totalPublicLinks, totalPrivateLinks);
                                                                           ^
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:91:75: warning: too many arguments for format [-Wformat-extra-args]
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:91:75: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:91:75: warning: too many arguments for format [-Wformat-extra-args]
src/main.cpp:93:80: warning: unknown conversion type character 'z' in format [-Wformat=]
             fprintf(stderr, "Detected %zu nodes in cycles\n", NodesWithCycles());
                                                                                ^
src/main.cpp:93:80: warning: too many arguments for format [-Wformat-extra-args]
src/main.cpp:93:80: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:93:80: warning: too many arguments for format [-Wformat-extra-args]
src/main.cpp: In function 'int main(int, char**)':
src/main.cpp:275:72: warning: unknown conversion type character 'z' in format [-Wformat=]
             printf("Found %zu ambiguous includes\n\n", ambiguous.size());
                                                                        ^
src/main.cpp:275:72: warning: too many arguments for format [-Wformat-extra-args]
src/main.cpp:275:72: warning: unknown conversion type character 'z' in format [-Wformat=]
src/main.cpp:275:72: warning: too many arguments for format [-Wformat-extra-args]
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/Component.o src/Component.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/Configuration.o src/Configuration.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/generated.o src/generated.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/Input.o src/Input.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/Output.o src/Output.cpp -std=c++11 -O3
src/Output.cpp: In function 'void PrintLinksForTarget(Component*)':
src/Output.cpp:178:57: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("Public linked (%zu):", sortedPubLinks.size());
                                                         ^
src/Output.cpp:178:57: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:178:57: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:178:57: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:184:61: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("\nPrivate linked (%zu):", sortedPrivLinks.size());
                                                             ^
src/Output.cpp:184:61: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:184:61: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:184:61: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp: In function 'void PrintInfoOnTarget(Component*)':
src/Output.cpp:199:44: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("Lines of Code: %zu\n", c->loc());
                                            ^
src/Output.cpp:199:44: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:199:44: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:199:44: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:204:63: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("Public dependencies (%zu): ", sortedPubDeps.size());
                                                               ^
src/Output.cpp:204:63: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:204:63: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:204:63: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:209:66: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("\nPrivate dependencies (%zu):", sortedPrivDeps.size());
                                                                  ^
src/Output.cpp:209:66: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:209:66: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:209:66: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:217:45: warning: unknown conversion type character 'z' in format [-Wformat=]
     printf("\nFiles (%zu):", c->files.size());
                                             ^
src/Output.cpp:217:45: warning: too many arguments for format [-Wformat-extra-args]
src/Output.cpp:217:45: warning: unknown conversion type character 'z' in format [-Wformat=]
src/Output.cpp:217:45: warning: too many arguments for format [-Wformat-extra-args]
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/CmakeRegen.o src/CmakeRegen.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -c -Wall -Wextra -Wpedantic -o obj/Analysis.o src/Analysis.cpp -std=c++11 -O3
/c/msys64/mingw64/bin/g++.exe -o cpp-dependencies obj/main.o obj/Component.o obj/Configuration.o obj/generated.o obj/Input.o obj/Output.o obj/CmakeRegen.o obj/Analysis.o -lboost_filesystem-mt -lboost_system-mt -O3
```

After compilation, the output of `cpp-dependencies.exe" --stats` is:

`zu components with zu public dependencies, zu private dependencies
Detected zu nodes in cycles`

I compiled it using the following compiler:

```
$ /c/msys64/mingw64/bin/g++.exe --version
g++.exe (Rev1, Built by MSYS2 project) 6.1.0
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by cwfitzgerald** [comment]
Question, why aren't these using std::cout? There's no portable way of printing size_t (as it could be 32bit or 64 bit) and no portable way of printing long longs without using gnu extentions. Standard streams would solve this issue. The other way would be to use crap tons of preprocessor logic, but I don't think any of us want that littering our code.

**Comment by dascandy** [comment]
Closing as your fix for #11 also fixed this one. Thanks!
